drivers/tapdisk-image.c

```diff
@@ -287,6 +289,52 @@ tapdisk_image_close_chain(struct list_head *list)
 		tapdisk_image_close(image);
 }
 
+static int
+find_parent_nbd_path(int prt_devnum, char **prt_nbd_path)
+{
+	int err;
+	char *pattern;
+	glob_t glbuf = { 0 };
+
+	/* This is glob pattern not a regex */
+	err = asprintf(&pattern, "%s/nbd[0-9]*.%d", BLKTAP2_CONTROL_DIR, prt_devnum);
+	if (err == -1) {
+		errno = ENOMEM;
+		return err;
+	}
+
+	err = glob(pattern, 0, NULL, &glbuf);
+	switch (err) {
+	case GLOB_NOMATCH:
+		EPRINTF("%s, failed to find parent NBD path", pattern);
```

Comment:
What's the return value of this function? Above it returns -1 and sets errno, here it returns a value from the `GLOB_*` namespace, and below it returns -errno.